code/modules/mob/dead/observer/observer.dm

```diff
@@ -495,13 +495,18 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	updateghostimages()
 
 /proc/updateallghostimages()
+	remove_nulls(ghost_images_full)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/code/__HELPERS/lists.dm#L106

Basically it creates an empty list with list.len lenght, this means it has list.len nulls in it, then it minus that from the original list.